src/components/IGVFileSelector.js

```diff
@@ -1,131 +1,105 @@
 import _ from 'lodash/fp'
-import { Component } from 'react'
+import { useState } from 'react'
 import { div, h, label } from 'react-hyperscript-helpers'
 import { AutoSizer, List } from 'react-virtualized'
 import ButtonBar from 'src/components/ButtonBar'
-import { ButtonPrimary, Clickable, IdContainer, LabeledCheckbox, Link, Select } from 'src/components/common'
+import { ButtonPrimary, IdContainer, LabeledCheckbox, Link, Select } from 'src/components/common'
 import * as Style from 'src/libs/style'
 import * as Utils from 'src/libs/utils'
 
 
-const styles = {
-  columnName: {
-    paddingLeft: '0.25rem',
-    flex: 1, display: 'flex', alignItems: 'center',
-    ...Style.noWrapEllipsis
-  }
-}
-
 const getStrings = v => {
   return Utils.cond(
     [_.isString(v), () => [v]],
-    [v && v.items, () => _.map(getStrings, v.items)],
+    [!!v?.items, () => _.map(getStrings, v.items)],
     () => []
   )
 }
 
 const MAX_CONCURRENT_IGV_FILES = 10
 
-export class IGVFileSelector extends Component {
-  constructor(props) {
-    super(props)
-    this.state = {
-      selectedFiles: _.fromPairs(_.map(v => [v, false], this.getIGVFileList())),
-      refGenome: 'hg38'
-    }
-  }
-
-  toggleVisibility(name) {
-    this.setState(_.update(['selectedFiles', name], b => !b))
-  }
-
-  getIGVFileList() {
-    const { selectedEntities } = this.props
-    return _.flow(
+const IGVFileSelector = ({ selectedEntities, onSuccess }) => {
+  const [refGenome, setRefGenome] = useState('hg38')
+  const [selections, setSelections] = useState(() => {
+    const allFiles = _.flow(
       _.flatMap(row => _.flatMap(getStrings, row.attributes)),
-      _.uniq,
-      _.filter(v => /\.(cram|bam|bed|vcf)$/.test(v))
+      _.uniq
     )(selectedEntities)
-  }
 
-  getSelectedFilesList() {
-    const { selectedFiles } = this.state
-    return _.flow(
-      _.keys,
-      _.filter(v => selectedFiles[v])
-    )(selectedFiles)
-  }
+    return _.reduce((acc, filePath) => {
+      const [, base, extension] = /(.+)\.([^.]+)$/.exec(filePath) || []
 
-  buttonIsDisabled() {
-    const selectedFilesList = this.getSelectedFilesList()
-    return !(selectedFilesList.length > 0 && selectedFilesList.length <= MAX_CONCURRENT_IGV_FILES)
-  }
+      if (!base) return acc
 
-  setAll(value) {
-    this.setState({ selectedFiles: _.fromPairs(_.map(v => [v, value], this.getIGVFileList())) })
-  }
+      const matchingIndex = Utils.switchCase(extension,
```

Comment:
TOL: Just making sure I understand the logic here. If we detect a known index file pattern, we'll explicitly pass that. If we don't, we'll pass `undefined`, which means IGV will infer based on the existing logic. So this will continue to work if e.g. the data didn't mention the index file explicitly, but it did exist at the inferred path.